fedimint-api-client/src/api.rs

```diff
@@ -287,6 +300,28 @@ pub trait FederationApiExt: IRawFederationApi {
         }
     }
 
+    /// Like [`Self::request_single_peer`], but API more like
+    /// [`Self::request_with_strategy`].
+    async fn request_single_peer_federation<FedRet>(
+        &self,
+        timeout: Option<Duration>,
```

Comment:
Why have an optional timeout here? If we want timeouts these should likely be added on externally. We could implement an extension trait that would let us do:

```rust
some_future
    .timeout(Duration::from_secs(1))
    .await
    .expect("timed out");
```